localstack/services/awslambda/provider.py

```diff
@@ -2362,12 +2573,50 @@ def list_layer_versions(
         max_items: MaxLayerListItems = None,
         compatible_architecture: Architecture = None,
     ) -> ListLayerVersionsResponse:
-        ...
+
+        validation_errors = api_utils.validate_layer_runtimes_and_architectures(
+            [compatible_runtime] if compatible_runtime else [],
+            [compatible_architecture] if compatible_architecture else [],
+        )
+        if validation_errors:
+            raise ValidationException(
+                f"{len(validation_errors)} validation error{'s' if len(validation_errors) > 1 else ''} detected: {';'.join(validation_errors)}"
+            )
+
+        # TODO: handle filter: compatible_runtime
+        # TODO: handle filter: compatible_architecture
+        state = lambda_stores[context.account_id][context.region]
+        all_layer_versions = []
+        for layer in state.layers.values():
+            if (
+                layer_name in layer.arn
```

Comment:
Is this really a substring match? Can I list the layer versions of the layer "arn" and get the first one?
The API spec says its the name or arn, could we just extract the name and access by its key?